Makefile

```diff
@@ -1,16 +1,15 @@
-CXX		?= g++
+CXX		= g++-12
```

Comment:
That'll bite if others don't use the exact same distro and packaging as you. Best not to hard-assign the compiler in a Makefile.

Continue to use `?=` so the end-user can override (with, say, `g++-12`) in an env var.

Reply:
`-march=sapphirerapids` is supported only with g++-12 and above and adding conditional compile in Make is non trivial, at least to me :) The main build system is Meson anyway which handles this relatively easy.

Reply:
Not just you 😁 . "Read the docs, provide path to compiler that support the following flags" is about all you can expect with a Makefile.